error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/JUnitClassDeclarationTest.java

```diff
@@ -0,0 +1,99 @@
+package tech.picnic.errorprone.bugpatterns;
+
+import com.google.errorprone.BugCheckerRefactoringTestHelper;
+import com.google.errorprone.BugCheckerRefactoringTestHelper.TestMode;
+import com.google.errorprone.CompilationTestHelper;
+import org.junit.jupiter.api.Test;
+
+final class JUnitClassDeclarationTest {
+  private final CompilationTestHelper compilationTestHelper =
+      CompilationTestHelper.newInstance(JUnitClassDeclaration.class, getClass());
+  private final BugCheckerRefactoringTestHelper refactoringTestHelper =
+      BugCheckerRefactoringTestHelper.newInstance(JUnitClassDeclaration.class, getClass());
+
+  @Test
+  void identification() {
+    compilationTestHelper
+        .addSourceLines(
+            "A.java",
+            "import org.junit.jupiter.api.Test;",
+            "",
+            "// BUG: Diagnostic contains:",
+            "class A {",
+            "  @Test",
+            "  void foo() {}",
+            "}")
+        .addSourceLines(
+            "B.java",
+            "import org.junit.jupiter.params.ParameterizedTest;",
+            "",
+            "// BUG: Diagnostic contains:",
+            "class B {",
+            "  @ParameterizedTest",
```

Comment:
Is there a specific reason to use `@ParameterizedTest`? To also account for that case?

Reply:
Yes, we check whether the class contains methods annotated with `@Test` or with _meta_ annotation `org.junit.jupiter.api.TestTemplate`. `@ParameterizedTest` is annotated with `@TestTemplate` and this test ensures we also match tests like this.